src/main/java/at/hannibal2/skyhanni/features/fame/UpgradeReminder.kt

```diff
@@ -0,0 +1,193 @@
+        set(value) {
+            ProfileStorageData.profileSpecific?.communityShopUpgrade = value
+        }
+
+    private var currentAccountUpgrade: CommunityShopUpgrade?
+        get() = ProfileStorageData.playerSpecific?.communityShopUpgrade
+        set(value) {
+            ProfileStorageData.playerSpecific?.communityShopUpgrade = value
+        }
+
+    private var inInventory = false
+    private var clickedUpgradeType: UpgradeType? = null
+    private var clickedUpgrade: CommunityShopUpgrade? = null
+    private var lastReminderSend = SimpleTimeMark.farPast()
+
+    // TODO: (for 0.27) merge this logic with reminder manager
+    @SubscribeEvent
+    fun onSecondPassed(event: SecondPassedEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+        if (!isEnabled()) return
+        if (ReminderUtils.isBusy()) return
+        if (inInventory || LorenzUtils.skyBlockArea == "Community Center") return
+        if (lastReminderSend.passedSince() < 30.seconds) return
+
+        currentProfileUpgrade?.sendReminderIfClaimable()
+        currentAccountUpgrade?.sendReminderIfClaimable()
+
+        lastReminderSend = SimpleTimeMark.now()
+    }
+
+    @SubscribeEvent
+    fun onInventoryOpen(event: InventoryFullyOpenedEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+        inInventory = event.inventoryName == "Community Shop"
+    }
+
+    @SubscribeEvent
+    fun onInventoryClose(event: InventoryCloseEvent) {
+        inInventory = false
+    }
+
+    @SubscribeEvent
+    fun onSlotClick(event: GuiContainerEvent.SlotClickEvent) {
+        if (!inInventory) return
+        val item = event.item ?: return
+        clickedUpgradeType = UpgradeType.fromItem(item) ?: return
+        clickedUpgrade = CommunityShopUpgrade.fromItem(item)
+    }
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+        if (upgradeStartedPattern.matches(event.message)) {
+            clickedUpgrade?.start()
+            when (clickedUpgradeType) {
+                UpgradeType.PROFILE -> currentProfileUpgrade = clickedUpgrade
+                UpgradeType.ACCOUNT -> currentAccountUpgrade = clickedUpgrade
+                null -> {}
+            }
+            return
+        }
+
+        upgradeClaimedPattern.matchMatcher(event.message) {
+            val claimedUpgradeName = group("upgrade")
+            when (claimedUpgradeName) {
+                currentProfileUpgrade?.name -> currentProfileUpgrade = null
+                currentAccountUpgrade?.name -> currentAccountUpgrade = null
+            }
+        }
+    }
+
+    private fun isEnabled() = SkyHanniMod.feature.misc.accountUpgradeReminder
+
+    fun disable() {
+        SkyHanniMod.feature.misc.accountUpgradeReminder = false
+    }
+
+    @SubscribeEvent
+    fun onConfigLoad(event: ConfigLoadEvent) {
+        val playerStorage = ProfileStorageData.playerSpecific ?: return
+        val name = playerStorage.currentAccountUpgrade ?: return
```

Comment:
We should use the `config migration fix event` instead. Then, we can remove the values `currentAccountUpgrade` and `nextAccountUpgradeCompletionTime` entirely